src/smc-webapp/editor_chat.cjsx

```diff
@@ -829,6 +845,9 @@ ChatRoom = (name) -> rclass
             foreground         : true
             foreground_project : true
 
+    render_realtime_typing_button: ->
+        <Button onClick={=> @setState(realtime_typing:!@state['realtime_typing'],realtime_typing_message_id:'')}>{if @state['realtime_typing'] then "Don't show" else "Show"} what I{"'"}m typing to others</Button>
```

Comment:
Since you already wrapped it in a function, you can easily increase readability of this Button by moving the if statement before returning the component.
ie.
```
if @state.realtime_typing
  text = "Don't show what I'm typing to others"
else
  text = "Show what I'm typing to others"
<Button onClick... >
  {text}
</Button>